assets/src/components/v2/disruption_diagram/disruption_diagram.tsx

```diff
@@ -717,81 +758,82 @@ const DisruptionDiagram: ComponentType<DisruptionDiagramData> = (props) => {
   // To calculate the height of that missing part, that is:
   // LINE_HEIGHT*scaleFactor/2 - MAX_ENDPOINT_HEIGHT*scaleFactor/2 + (hasEmphasis ? EMPHASIS_PADDING_TOP * scaleFactor : 0)
 
-  // offset is parent container height minus all the stuff below the very top of the line diagram
+  // So finally, the vertical viewBoxOffset is parent container height minus
+  // all the stuff below the very top of the line diagram
   const viewBoxOffset = 
-    height
-    - (LINE_HEIGHT * scaleFactor) / 2
-    - (MAX_ENDPOINT_HEIGHT * scaleFactor) / 2
+    originalHeight
+    - LINE_HEIGHT * scaleFactor / 2
+    - MAX_ENDPOINT_HEIGHT * scaleFactor / 2
     + (hasEmphasis ? EMPHASIS_PADDING_TOP * scaleFactor : 0)
 
   return (
-    <svg
-      // viewBoxOffset will always be > 0 by the time it's visible, but the console will
-      // still log an error if it's a negative number when it's not-yet-visible
-      viewBox={`0 ${-viewBoxOffset} 904 ${height + getEmphasisHeight(scaleFactor)}`}
-      transform={`translate(${translateX})`}
-      visibility={isDone ? "visible" : "hidden"}
-    >
-      <g transform={`translate(${L * scaleFactor} 0)`}>
-        <g id="line-map" transform={`scale(${scaleFactor})`}>
-          {effect !== "station_closure" && (
-            <EffectBackgroundComponent
-              effectRegionSlotIndexRange={props.effect_region_slot_index_range}
-              effect={effect}
+    <div style={{width: "100%", height: "100%"}} id="diagram-container" ref={ref}>
+      <svg
+        viewBox={`0 ${-viewBoxOffset} ${DIAGRAM_WIDTH} ${originalHeight + getEmphasisHeight(scaleFactor)}`}
+        transform={`translate(${translateX})`}
+        visibility={isDone ? "visible" : "hidden"}
+      >
+        <g transform={`translate(${L * scaleFactor} 0)`}>
+          <g id="line-map" transform={`scale(${scaleFactor})`} ref={measuredRef}>
```

Comment:
What do you think about naming the ref callback to something like `measureLineMapNode`?